File: src/daily/_202208/_20220825/Daily.java
package daily._202208._20220825;

import java.util.Arrays;
import java.util.List;

/**
 * @author John.h3
 * created on 2022/8/25
 */
public class Daily {
    class Solution {
        public List<Integer> findClosestElements(int[] arr, int k, int x) {
            var r = Arrays.binarySearch(arr, x);
            if (r < 0) {
                r = -(r + 1);
            }
            if (r < 0) {
                r++;
            } else if (r == arr.length) {
                r--;
            }
            var l = r - 1;
            while (r - 1 - l != k) {
                if (r < arr.length && l >= 0) {
                    if (Math.abs(arr[r] - x) < Math.abs(arr[l] - x) || (arr[r] - x == x - arr[l] && arr[r] < arr[l])) {
                        r++;
                    } else {
                        l--;
                    }
                } else if (l < 0) {
                    r++;
                } else {
                    l--;
                }
            }
            return Arrays.stream(arr, l + 1, r).boxed().toList();
        }
    }
}

File: src/daily/_202208/_20220826/Daily.kt
package daily._202208._20220826

/**
 * @author John.h3
 * created on 2022/8/26
 */

class Solution {
    fun maxProduct(nums: IntArray): Int {
        var m1 = nums[0]
        var m2: Int
        if (nums[1] > m1) {
            m1 = nums[1]
            m2 = nums[0]
        } else {
            m2 = nums[1]
        }
        for (i in 2 until nums.size) {
            if (nums[i] > m2) {
                if (nums[i] > m1) {
                    m2 = m1
                    m1 = nums[i]
                } else {
                    m2 = nums[i]
                }
            }
        }
        return (m1 - 1) * (m2 - 1)
    }
}